etc/ci/performance/runner.py

```diff
@@ -0,0 +1,263 @@
+                median_result[k] = group[0][k]
+            else:
+                try:
+                    median_result[k] = median([x[k] for x in group
+                                               if x[k] is not None])
+                except StatisticsError:
+                    median_result[k] = -1
+        median_results.append(median_result)
+    return median_results
+
+
+def save_result_json(results, filename, manifest, expected_runs):
+
+    results = filter_result_by_manifest(results, manifest)
+    results = take_result_median(results, expected_runs)
+
+    if len(results) == 0:
+        with open(filename, 'w') as f:
+            json.dump("No test result found in the log. All tests timeout?",
+                      f, indent=2)
+    else:
+        with open(filename, 'w') as f:
+            json.dump(results, f, indent=2)
+    print("Result saved to {}".format(filename))
+
+
+def format_result_summary(results):
+    failures = list(filter(lambda x: x['domComplete'] == -1, results))
+    result_log = """
+========================================
+Total {total} tests; {suc} succeeded, {fail} failed.
+
+Failure summary:
+""".format(
+        total=len(results),
+        suc=len(list(filter(lambda x: x['domComplete'] != -1, results))),
+        fail=len(failures)
+    )
+    uniq_failures = list(set(map(lambda x: x['testcase'], failures)))
+    for failure in uniq_failures:
+        result_log += " - {}\n".format(failure)
+
+    result_log += "========================================\n"
+
+    return result_log
+
+
+def main():
+    parser = argparse.ArgumentParser(
+        description="Run page load test on servo"
+    )
+    parser.add_argument("tp5_manifest",
+                        help="the test manifest in tp5 format")
+    parser.add_argument("output_file",
+                        help="filename for the output json")
+    parser.add_argument("--runs",
+                        type=int,
+                        default=20,
+                        help="number of runs for each test case. Defult: 20")
+    parser.add_argument("--timeout",
+                        type=int,
+                        default=300,  # 5 min
+                        help=("kill the test if not finished in time (sec)."
+                              " Default: 5 min"))
+    parser.add_argument("--engine",
+                        type=str,
+                        default='servo',
+                        help=("The engine to run the tests on. Currently only"
+                              " servo and gecko are supported."))
+    args = parser.parse_args()
+    if args.engine == 'servo':
+        command_factory = get_servo_command
+    elif args.engine == 'gecko':
+        command_factory = get_gecko_command
+    try:
+        # Assume the server is up and running
+        testcases = load_manifest(args.tp5_manifest)
+        results = []
+        for testcase in testcases:
+            command = command_factory(testcase, args.timeout)
```

Comment:
Since we're running both commands under `timeout` in the same way, let's handle wrapping the command with `timeout` here instead of inside `get_servo_command` and `get_gecko_command`.